### PR TITLE
upgrade: script record upgrade from v4 to v5

### PR DESCRIPTION
**Pull request by ppanero**
- closes https://github.com/inveniosoftware/invenio-app-rdm/issues/926

### CODE REVIEW

**Comment by ppanero** [comment]
Discussed IRL:

1- Flag check: Do you have subjects or affiliations with more than "name/subject". i.e. will they cause troubles?
2- If yes, give a recipe to fix them
3- If not (or after fix) run the script and assume:
    3.1 - id is present: is a vocabulary (note that affiliations will contain id+name in the dumped data)
    3.2 - name/subject is present: is s custom term

File: invenio_app_rdm/upgrade_scripts/migrate_4_0_to_6_0.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+                "contributors": inv_affs_contributors
+            }
+
+    invalid_affiliations = {}
+    if invalid_affiliations_rec:
+        invalid_affiliations["record"] = invalid_affiliations_rec
+    if invalid_affiliations_draft:
+        invalid_affiliations["draft"] = invalid_affiliations_draft
+
+    print(
```
Comment (zzacharo) [inline invenio_app_rdm/upgrade_scripts/migrate_4_0_to_6_0.py:115]: Let's refactor this part a bit to something like
```python
if invalid_affiliations:
        print(
            f"Your instance has {total} affiliations that need to be fixed. " +
            "Check the invalid_affiliation.yaml file for more details."
        )
        with open('invalid_affiliations.yaml', 'w') as f:
            yaml.dump(list(invalid_affiliations), f)
else:
        print(f"Your instance has no affiliations that need to be fixed.")
```

Comment (ppanero) [reply invenio_app_rdm/upgrade_scripts/migrate_4_0_to_6_0.py:115]: We should also keep a `ROR` counter to tell the user if they need to index the ROR vocabulary or they can do with only free text affiliations.

Comment (ppanero) [reply invenio_app_rdm/upgrade_scripts/migrate_4_0_to_6_0.py]: Done :)

File: invenio_app_rdm/upgrade_scripts/migrate_4_0_to_6_0.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+            "Check the invalid_affiliation.yaml file for more details."
+        )
+        with open('invalid_affiliations.yaml', 'w') as f:
+            yaml.dump(list(invalid_affiliations), f)
+    else:
+        print(f"All your instance's affiliations are valid.")
+    if needs_ror:
+        print(
+            "You have affiliations with ROR identifiers, you need to " +
+            "add its vocabulary. Instructions to do so are available in TODO"
```
Comment (zzacharo) [inline invenio_app_rdm/upgrade_scripts/migrate_4_0_to_6_0.py:131]: Note not to forget to replace the `TODO`

**Review by zzacharo — COMMENTED** [review]
LGTM. We just need to fill in the missing links!

**Pull request merged**